lib/terraform/libvirt_domain.pm

```diff
@@ -0,0 +1,133 @@
+# SUSE's openQA tests
+#
+# Copyright © 2019 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Base class for terraform domains (VMs)
+#
+# Maintainer: Jose Lausuch <jalausuch@suse.de>
+
+package terraform::libvirt_domain;
+use Mojo::Base -base;
+use testapi;
+use XML::Simple;
+use Data::Dumper;
+
+has domain_name   => undef;    # domain name
+has domain_ip     => undef;    # private IP of the VM
+has virtio_device => undef;    # host device that corresponds to virtio
+has host          => undef;    # host object where the VM is running
+
+use constant LOGIN_TIMEOUT => 30;    # (x 30)
+
+sub init {
+    my ($self) = @_;
+    $self->_set_virtio_device();
+    $self->_login();
+}
+
+
+=head2 login
+
+    login($cmd);
+
+Login to the VM on the console with the given user and password
+
+=cut
+sub _login {
+    my ($self, %args) = @_;
+    $args{user} //= 'root';
+    $args{pwd}  //= 'nots3cr3t';
+    my $timeout = LOGIN_TIMEOUT;
+    $self->host->run_ssh_command('cat ' . $self->virtio_device . " > out.log &");
+    $self->host->run_ssh_command('echo -e "\n" > ' . $self->virtio_device);    # trick for the login prompt to appear
+    while ($timeout > 0) {
+        my $prompt = $self->host->run_ssh_command("cat out.log");
+        record_info('PROMPT', $self->domain_name . " prompt :\n" . $prompt) if check_var('DEBUG', 1);
+        if ($prompt =~ /login/) {
+            $self->host->run_ssh_command('echo -e "' . $args{user} . '\n" > ' . $self->virtio_device);
+            sleep 10;
+            $self->host->run_ssh_command('echo -e "' . $args{pwd} . '\n" > ' . $self->virtio_device);
+            sleep 10;
+            last;
+        }
+        $timeout -= 1;
+        sleep 10;
+    }
+    my $pid = $self->host->run_ssh_command("pidof cat");
```

Comment:
If there are more `cat` running, you get a list and maybe kill the wrong one.
see line 46

Reply:
yes, and I've thought about this before and wanted to change it, but didn't found a better way.